### ISSUE TITLE
Controller may be leaking its view

### ISSUE THREAD

**Issue by AranHase** [bluelinelabs/Conductor#206]
I recently started getting lots of LeakCanary warnings about Android Views being leaked.

Here's the result of a quick investigation:

First I added this method override on the demo app:
```
// RefWatchingController.java (demo)
@Override
    protected void onDestroyView(@NonNull View view) {
        DemoApplication.refWatcher.watch(view);
        super.onDestroyView(view);
    }
```

Then, reverted Conductor back to a commit before this one (the one that started this issue): efbdf913bde0541d55a2d7d0546e58bfcf4887ea

Run the demo app and I can open/close samples and all is okay.

Then checkout the commit: efbdf913bde0541d55a2d7d0546e58bfcf4887ea

Re-run the demo app, and as soon as I change pages, the leaks start to show up. Here's one of them:

http://pastebin.com/y6ZEti2E

**Comment by EricKuck** [comment]
Thanks for the issue and the PR! Looks good.

**Comment by PaulWoitaschek** [comment]
Can we have a hotfix release? Memory leaks are devasting. @erickuck

**Comment by EricKuck** [comment]
Definitely. I was planning to push one within the next few hours. I'll comment back when it's done.

**Comment by EricKuck** [comment]
Alright, now published as `2.0.7`!
